src/Magento/FunctionalTestingFramework/Upgrade/UpdateTestSchemaPaths.php

```diff
@@ -29,56 +57,42 @@ class UpdateTestSchemaPaths implements UpgradeInterface
      */
     public function execute(InputInterface $input, OutputInterface $output)
     {
-        // @codingStandardsIgnoreStart
-        $relativeToUrn = [
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/DataGenerator/etc/dataOperation.xsd"
-                => "urn:magento:mftf:DataGenerator/etc/dataOperation.xsd",
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/DataGenerator/etc/dataProfileSchema.xsd"
-                => "urn:magento:mftf:DataGenerator/etc/dataProfileSchema.xsd",
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Page/etc/PageObject.xsd"
-                => "urn:magento:mftf:Page/etc/PageObject.xsd",
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Page/etc/SectionObject.xsd"
-                => "urn:magento:mftf:Page/etc/SectionObject.xsd",
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Test/etc/actionGroupSchema.xsd"
-                => "urn:magento:mftf:Test/etc/actionGroupSchema.xsd",
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Test/etc/testSchema.xsd"
-                => "urn:magento:mftf:Test/etc/testSchema.xsd",
-            "dev/tests/acceptance/vendor/magento/magento2-functional-testing-framework/src/Magento/FunctionalTestingFramework/Suite/etc/suiteSchema.xsd"
-                => "urn:magento:mftf:Suite/etc/suiteSchema.xsd"
-        ];
-        // @codingStandardsIgnoreEnd
-
-        $relativePatterns = [];
-        $urns = [];
-        // Prepare array of patterns to URNs for preg_replace (replace / to escapes
-        foreach ($relativeToUrn as $relative => $urn) {
-            $relativeReplaced = str_replace('/', '\/', $relative);
-            $relativePatterns[] = '/[.\/]+' . $relativeReplaced  . '/';
-            $urns[] = $urn;
-        }
-
-        $testsUpdated = 0;
+        $this->output = $output;
+        $this->testsUpdated = 0;
         $testPaths[] = $input->getArgument('path');
```

Comment:
When I pass in a `path` argument, it does not search that directory recursively.

Is this expected behavior?

Reply:
> @jilu1 Can you clean up the commit history on this branch? It looks weird. Can you rebase it on top of `MQE-1963` to clean it up?

I will clean up after MQE-1963 is finalized.

Reply:
When `path` is used. It suppose to search directory recursively for Test entities (Test, Page, Section, etc) though. What's your test case?

Reply:
I ran:

```mftf upgrade:tests /Users/treece/dev/mftf/magento2ce```

and it found nothing to upgrade. It doesn't search recursively.

Reply:
Does it work for you before?

Reply:
Yes.

Reply:
I debugged your code. And this array is empty:

```$xmlFiles = ScriptUtil::getModuleXmlFilesByScope($testPaths, $type);```

Reply:
I use this to upgrade individual test modules. I will check with the behavior in `develop`.

Reply:
What path did you pass in? If you passed in something like:

```/Users/treece/dev/mftf/magento2ce/app/code/Magento/Backend```

then maybe it works.
But I'm saying I expected this to work:

```/Users/treece/dev/mftf/magento2ce```

do you see the difference?

Reply:
If I remember correctly, with `develop` branch, `path` is not working for everything. That's why the `path` is changed to optional. For MQE-2013 to work, you need to have MQE-1963. I will compare and update here

Reply:
@tomreece Actually the `path` argument has changed its meaning. This is part of the refactory in MQE-1963. 
Previously we will search `path` with a pattern. For example, you can find all xml files, or all -meta.xml files. But there are two problems here. 1. Not all Test Modules are inside one single root directory. 2. Test and Suite xml files don't have special pattern.
Because of these limitation, I refactor the upgrade command and make `path` optional. 
To be clear, I also updated DevDocs.
Let me know if this makes sense.

Reply:
It makes sense. Thanks for clarifying.